api/http/handler/hostmanagement/openamt/openamt.go

```diff
@@ -0,0 +1,218 @@
+		}
+		if payload.DomainName == "" {
+			return errors.New("domain name must be provided")
+		}
+		if payload.CertFileText == "" {
+			return errors.New("certificate file must be provided")
+		}
+		if payload.CertPassword == "" {
+			return errors.New("certificate password must be provided")
+		}
+		if payload.UseWirelessConfig {
+			if payload.WifiAuthenticationMethod == "" {
+				return errors.New("wireless authentication method must be provided")
+			}
+			if payload.WifiEncryptionMethod == "" {
+				return errors.New("wireless encryption method must be provided")
+			}
+			if payload.WifiSSID == "" {
+				return errors.New("wireless config SSID must be provided")
+			}
+			if payload.WifiPskPass == "" {
+				return errors.New("wireless config PSK passphrase must be provided")
+			}
+		}
+	}
+
+	return nil
+}
+
+// @id OpenAMTConfigureDefault
+// @summary Enable OpenAMT capabilities
+// @description Enable OpenAMT capabilities
+// @description **Access policy**: administrator
+// @tags intel
+// @security jwt
+// @accept json
+// @produce json
+// @param body body openAMTConfigureDefaultPayload true "OpenAMT Settings"
+// @success 204 "Success"
+// @failure 400 "Invalid request"
+// @failure 403 "Permission denied to access settings"
+// @failure 500 "Server error"
+// @router /open_amt [post]
+func (handler *Handler) openAMTConfigureDefault(w http.ResponseWriter, r *http.Request) *httperror.HandlerError {
+	var payload openAMTConfigureDefaultPayload
+	err := request.DecodeAndValidateJSONPayload(r, &payload)
+	if err != nil {
+		logrus.WithError(err).Error("Invalid request payload")
+		return &httperror.HandlerError{StatusCode: http.StatusBadRequest, Message: "Invalid request payload", Err: err}
+	}
+
+	if payload.EnableOpenAMT {
+		certificateErr := validateCertificate(payload.CertFileText, payload.CertPassword)
+		if certificateErr != nil {
+			return &httperror.HandlerError{StatusCode: http.StatusBadRequest, Message: "Error validating certificate", Err: certificateErr}
+		}
+
+		err = handler.enableOpenAMT(payload)
+		if err != nil {
+			return &httperror.HandlerError{StatusCode: http.StatusBadRequest, Message: "Error enabling OpenAMT", Err: err}
+		}
+		return response.Empty(w)
+	}
+
+	err = handler.disableOpenAMT()
+	if err != nil {
+		return &httperror.HandlerError{StatusCode: http.StatusBadRequest, Message: "Error disabling OpenAMT", Err: err}
+	}
+	return response.Empty(w)
+}
+
+func validateCertificate(certificateRaw string, certificatePassword string) error {
+	certificateData, err := base64.StdEncoding.Strict().DecodeString(certificateRaw)
+	if err != nil {
+		return err
+	}
+
+	_, certificate, _, err := pkcs12.DecodeChain(certificateData, certificatePassword)
+	if err != nil {
+		return err
+	}
+
+	if certificate == nil {
+		return errors.New("certificate could not be decoded")
+	}
+
+	issuer := certificate.Issuer.CommonName
+	if !isValidIssuer(issuer) {
+		return fmt.Errorf("certificate issuer is invalid: %v", issuer)
```

Comment:
can we please tell the user how the cert is invalid - (ie, tell them who the issuer is, and what limited set we are testing for)
Also, consider that this is the list today, but a new device may have others - `rpic amtinfo -cert` tells me:

```
sven@p1:~/src/open-amt-cloud-toolkit/rpc-go$ sudo ./rpc amtinfo -cert
[sudo] password for sven: 
Certificate Hashes	:
Go Daddy Class 2 CA (Default,Active)
   SHA256: c3846bf24b9e93ca64274c0ec67c1ecc5e024ffcacd2d74019350e81fe546ae4
Go Daddy Root CA-G2 (Default,Active)
   SHA256: 45140b3247eb9cc8c5b4f0d7b53091f73292089e6e5a63e2749dd3aca9198eda
Comodo AAA CA (Default,Active)
   SHA256: d7a7a0fb5d7e2731d771e9484ebcdef71d5f0c3e0a2948782bc83ee0ea699ef4
Starfield Class 2 CA (Default,Active)
   SHA256: 1465fa205397b876faa6f0a9958e5590e40fcc7faa4fb7c2c8677521fb5fb658
Starfield Root CA-G2 (Default,Active)
   SHA256: 2ce1cb0bf9d2f9e102993fbe215152c3b2dd0cabde1c68e5319b839154dbb7f5
VeriSign Class 3 Primary CA-G5 (Default,Active)
   SHA256: 9acfab7e43c8d880d06b262a94deeee4b4659989c3d0caf19baf6405e41ab7df
Baltimore CyberTrust Root (Default,Active)
   SHA256: 16af57a9f676b0ab126095aa5ebadef22ab31119d644ac95cd4b93dbf3f26aeb
Cybertrust Global Root (Default,Active)
   SHA256: 960adf0063e96356750c2965dd0a0867da0b9cbd6e77714aeafb2349ab393da3
Verizon Global Root (Default,Active)
   SHA256: 68ad50909b04363c605ef13581a939ff2c96372e3f12325b0a6861e1d59f6603
Entrust.net CA (2048) (Default,Active)
   SHA256: 6dc47172e01cbcb0bf62580d895fe2b8ac9ad4f873801e0c10b9c837d21eb177
Entrust Root CA (Default,Active)
   SHA256: 73c176434f1bc6d5adf45b0e76e727287c8de57616c1e6e6141a2b2cbc7d8e4c
Entrust Root CA-G2 (Default,Active)
   SHA256: 43df5774b03e7fef5fe40d931a7bedf1bb2e6b42738c4e6d3841103d3aa7f339
VeriSign Universal Root CA (Default,Active)
   SHA256: 2399561127a57125de8cefea610ddf2fa078b5c8067f4e828290bfb860e84b3c
Affirm Trust Premium (Default,Active)
   SHA256: 70a73f7f376b60074248904534b11482d5bf0e698ecc498df52577ebf2e93b9a
DigiCert Global Root CA (Default,Active)
   SHA256: 4348a0e9444c78cb265e058d5e8944b4d84f9662bd26db257f8934a443c70161
DigiCert Global Root G2 (Default,Active)
   SHA256: cb3ccbb76031e5e0138f8dd39a23f9de47ffc35e43c1144cea27d46a5ab1cb5f
DigiCert Global Root G3 (Default,Active)
   SHA256: 31ad6648f8104138c738f39ea4320133393e3a18cc02296ef97c2ac9ef6731d0
DigiCert Trusted Root G4 (Default,Active)
   SHA256: 552f7bdcf1a7af9e6ce672017f4f12abf77240c78e761ac203d1d9d20ac89988
GlobalSign NP RSA CA 2018 (Default,Active)
   SHA256: 67540a47aa5b9f34570a99723cfefa96a96ee3f0d9b8bf4def9440b8065d665d
GlobalSign NP ECC CA 2018 (Default,Active)
   SHA256: 7224395222cd588c4f2683716922addb41e39b581ac34fa87b39efa896fbb39e
GlobalSign Root CA - R3 (Default,Active)
   SHA256: cbb522d7b7f127ad6a0113865bdf1cd4102e7d0759af635a7cf4720dc963c53b
GlobalSign ECC Root CA - R5 (Default,Active)
   SHA256: 179fbc148a3dd00fd24ea13458cc43bfa7f59c8182d783a513f6ebec100c8924
GlobalSign Root CA - R6 (Default,Active)
   SHA256: 2cabeafe37d06ca22aba7391c0033d25982952c453647349763a3ab5ad6ccf69
```

Reply:
as compared to on my "should be tool old" v6.1.10 (and intentionally vulnerable to hacks)

```
sys-admin@ubuntu-9e11c137cf:~$ sudo docker run --rm -it --device /dev/mei0 --pull always ptrrd/openamt:rpc-go amtinfo -cert
rpc-go: Pulling from ptrrd/openamt
Digest: sha256:5b3c1764915a0296333ea9d49c48edd39b05c5c80d2beaf7796945e9125f48a9
Status: Image is up to date for ptrrd/openamt:rpc-go
Certificate Hashes	:
VeriSign Class 3 Primary CA-G1 (Default,Active)
   SHA1: 742c3192e607e424eb4549542be1bbc53e6174e2
VeriSign Class 3 Primary CA-G3 (Default,Active)
   SHA1: 132d0d45534b6997cdb2d5c339e25576609b5cc6
Go Daddy Class 2 CA (Default,Active)
   SHA1: 2796bae63f1801e277261ba0d77770028f20eee4
Comodo AAA CA (Default,Active)
   SHA1: d1eb23a46d17d68fd92564c2f1f1601764d8e349
Starfield Class 2 CA (Default,Active)
   SHA1: ad7e1c28b064ef8f6003402014c3d0e3370eb58a
VeriSign Class 3 Primary CA-G2 (Default,Active)
   SHA1: 85371ca6e550143dce2803471bde3a09e8f8770f
VeriSign Class 3 Primary CA-G1.5 (Default,Active)
   SHA1: a1db6393916f17e4185509400415c70240b0ae6b
VeriSign Class 3 Primary CA-G5 (Default,Active)
   SHA1: 4eb6d578499b1ccf5f581ead56be3d9b6744a5e5
```

Reply:
I don't think we should be validating this at all, but rather let the MPS server validate it.  We know currently there is a list of 4 valid issuers but that could change in the future.  We don't want to be repeating what the MPS server already I think probably does?

Reply:
either way, tell the user more details - enough so they have a chance of realising what is wrong and what to google to fix it

Reply:
We could remove this validation, I added it mostly because I noticed that at this stage of the flow Intel doesn't validate the file, but rather does so in the activation step, and if I remember correctly with some very non-user friendly error messages. imo it's better to validate at this point and not allow the user to continue until the cert is valid.

For the UI I added this tooltip to the form, and in case of error we show the decoding error directly to the user
![image](https://user-images.githubusercontent.com/16310914/143589856-e39290b6-6c72-4894-91da-231d3ea57759.png)

![image](https://user-images.githubusercontent.com/16310914/143590045-997364ad-e8c8-44f3-a339-3dba86681a0f.png)

Reply:
Ok that's interesting.   In that case maybe we should leave it in.